spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -93,7 +94,8 @@
 
 	private final RedisTemplate<String, RedisLock> redisTemplate;
 
-	private final ThreadLocal<List<RedisLock>> threadLocks = new ThreadLocal<List<RedisLock>>();
+	private final ThreadLocal<Set<RedisLock>> weakThreadLocks = new ThreadLocal<Set<RedisLock>>();
+	private final ThreadLocal<List<RedisLock>> hardThreadLocks = new ThreadLocal<List<RedisLock>>();
```

Comment:
Not sure if I am correct or if I'm missing something, but it looks for me that we don't need this `ThreadLocal` stuff at all.
The `RedisLock` has `lockKey` as well as `thread` properties. Since it is normal `ReentrantLock`, we can use them exclusively only within the single `Thread`, hence I figure out this solution:
```
private final Set<RedisLock> locks = Collections.newSetFromMap(new WeakHashMap<RedisLock, Boolean>());
```
and iterate over them for `lockKey` and  
```
if (Thread.currentThread() != redisLock.thread) {
    throw new IllegalMonitorStateException();
}
```
Or even without the last one, because we have a locking guarantee with `localLock` and `WATCH` with `MULTI`.

WDYT ?